### ISSUE TITLE
How to get gpu global rank in the dataloader model?

### ISSUE THREAD

**Issue by FakeEnd** [Lightning-AI/pytorch-lightning#19331]
### Description & Motivation

My custom dataset using a numpy sampler to get data. If I cannot use different seed for different GPU/nodes, it will use the same seed forked from main process, which will sample the same data. So, I want to get global rank in `LightningDataModule.train_dataloader` to create different dataloader for different GPU/nodes. I see there is a local rank property in [here](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.core.LightningModule.html#lightning.pytorch.core.LightningModule.local_rank), but when I try to use `self.local_rank` to get it, it fails.

cc @borda

**Comment by ssharpe42** [comment]
@awaelchli when is `self.trainer` set in the data module?
